ir/src/main/scala/org/scalajs/ir/Names.scala

```diff
@@ -351,8 +369,8 @@ object Names {
 
     def apply(simpleName: SimpleMethodName, paramTypeRefs: List[TypeRef],
         resultTypeRef: TypeRef, isReflectiveProxy: Boolean): MethodName = {
-      if ((simpleName.isConstructor || simpleName.isStaticInitializer) &&
-          resultTypeRef != VoidRef) {
+      if ((simpleName.isConstructor || simpleName.isStaticInitializer ||
+          simpleName.isClassInitializer) && resultTypeRef != VoidRef) {
```

Comment:
Somewhat unrelated: But if we enforce the result type, why not the params?

Reply:
That said, if we change something I would go in the other direction, and *not* enforce the result type in `Names` at all. We can do that in the IR checker instead, which would speed things up a bit since we wouldn't have to check that every time we build a constructor name. There's nothing fundamental in Names that requires that a constructor has a `void` result type; it's more a question of IR validity.

WDYT?

Reply:
Yes, sounds reasonably. In fact, it seems that right now, nothing is checking that :-/

Reply:
Same for the reflexive proxy result type for that matter.